moderation/punishments.go

```diff
@@ -439,24 +440,56 @@ func AddMemberMuteRole(config *Config, id int64, currentRoles []int64) (removedR
 }
 
 func RemoveMemberMuteRole(config *Config, id int64, currentRoles []int64, mute MuteModel) (err error) {
+	newMemberRoles := decideUnmuteRoles(config, currentRoles, mute)
+	err = common.BotSession.GuildMemberEdit(config.GuildID, id, newMemberRoles)
+	return
+}
+
+func decideUnmuteRoles(config *Config, currentRoles []int64, mute MuteModel) []string {
+	newMemberRoles := make([]string, 0)
+
+	gs := bot.State.Guild(true, config.GuildID)
+	botState, err := bot.GetMember(gs.ID, common.BotUser.ID)
 
-	newMemberRoles := make([]string, 0, len(currentRoles)+len(config.MuteRemoveRoles))
+	gs.RLock()
+	defer gs.RUnlock()
+
+	guildRoles := make([]int64, len(gs.Guild.Roles))
+	for k, e := range gs.Guild.Roles {
+		guildRoles[k] = e.ID
+	}
+
+	if err != nil || botState == nil { // We couldn't find the bot on state, so keep old behaviour
+		for _, r := range currentRoles {
+			if r != config.IntMuteRole() {
+				newMemberRoles = append(newMemberRoles, strconv.FormatInt(r, 10))
+			}
+		}
+
+		for _, r := range mute.RemovedRoles {
+			if !common.ContainsInt64Slice(currentRoles, r) && common.ContainsInt64Slice(guildRoles, r) {
+				newMemberRoles = append(newMemberRoles, strconv.FormatInt(r, 10))
+			}
+		}
+
+		return newMemberRoles
+	}
+
+	yagHighest := bot.MemberHighestRole(gs, botState)
 
 	for _, v := range currentRoles {
-		if v != config.IntMuteRole() {
+		if v != config.IntMuteRole() && dutil.IsRoleAbove(yagHighest, gs.Role(false, v)) {
```

Comment:
If the user already had a role above the bot, wont this attempt to remove it (and therefor fail)?

Reply:
Oh, you are right! Will push a fix soon

Reply:
This should be fixed now